aten/src/ATen/core/MT19937RNGEngine.h

```diff
@@ -0,0 +1,177 @@
+#pragma once
+
+// define constants like M_PI and C keywords for MSVC
```

Comment:
I need a reminder why we reimplemented `std::mt19937`; I feel like I've asked you about this before, a comment in the header will make it official.

Reply:
That's because `at::mt19937` turns out to be faster in the pytorch codebase. Here are the results from the bechmark: https://github.com/syed-ahmed/benchmark-rngs. 
```
with -O2
Time to get 100000000 philox randoms with at::uniform_real_distribution = 0.462759s
X value is 0.128836
Time to get 100000000 at::mt19937 randoms with at::uniform_real_distribution = 0.39628s
X value is 0.365931
Time to get 100000000 std::mt19937 randoms with std::uniform_real_distribution = 0.352087s
X value is 0.365931
Time to get 100000000 std::mt19937 randoms with at::uniform_real_distribution = 0.419454s
X value is 0.365931
```
`std::mt19937` is faster when used in conjuction with `std::uniform_real_distribution`, however we can't use `std::uniform_real_distribution` because of this bug: http://open-std.org/JTC1/SC22/WG21/docs/lwg-active.html#2524. Plus, even if I used `std::uniform_real_distribution` and filtered out the 1's, it is a different algorithm than what's in pytorch currently and that messes up the tests... The other option, using `std::mt19937` with `at::uniform_real_distribution` is a tad bit slower than `at::mt19937` with `at::uniform_real_distribution` and hence, went with the latter. I can include this in the comment.

Reply:
Yes please!!